transport/transport.test.ts

```diff
@@ -536,6 +537,153 @@ describe.each(testMatrix())(
+      addPostTestCleanup(async () => {
+        await cleanupTransports([clientTransport, serverTransport]);
+      });
+
+      await waitFor(() => expect(clientTransport.connections.size).toEqual(1));
+      await waitFor(() => expect(serverTransport.connections.size).toEqual(1));
+
+      const oldClientSession = serverTransport.sessions.get('client');
+      const oldServerSession = clientTransport.sessions.get('SERVER');
+      expect(oldClientSession).toMatchObject({
+        id: oldServerSession?.advertisedSessionId,
+      });
+      expect(oldServerSession).toMatchObject({
+        id: oldClientSession?.advertisedSessionId,
+      });
+
+      clientTransport.reconnectOnConnectionDrop = false;
+      clientTransport.connections.forEach((conn) => conn.close());
+      await waitFor(() => expect(clientTransport.connections.size).toEqual(0));
+      await waitFor(() => expect(serverTransport.connections.size).toEqual(0));
+
+      clientTransport.reconnectOnConnectionDrop = true;
+      await clientTransport.connect('SERVER');
+      await waitFor(() => expect(clientTransport.connections.size).toEqual(1));
+      await waitFor(() => expect(serverTransport.connections.size).toEqual(1));
+
+      // expect new sessions to have been created
+      const newClientSession = serverTransport.sessions.get('client');
+      const newServerSession = clientTransport.sessions.get('SERVER');
+      expect(newClientSession).not.toBe(oldClientSession);
+      expect(newServerSession).not.toBe(oldServerSession);
+
+      await testFinishesCleanly({
+        clientTransports: [clientTransport],
+        serverTransport,
+      });
+    });
+  },
+);
+
+describe.each(testMatrix())(
+  'transport handshake grace period tests',
+  async ({ transport, codec }) => {
+    const opts: TransportOptions = {
+      codec: codec.codec,
+      heartbeatIntervalMs: 1_000,
+      heartbeatsUntilDead: 2,
+      sessionDisconnectGraceMs: 10_000,
+      // setting session grace to be higher so that only handshake grace passes
+      handshakeGraceMs: 5_000,
+    };
+    let testHelpers: TestSetupHelpers;
+    let getClientTransport: TestSetupHelpers['getClientTransport'];
+    let getServerTransport: TestSetupHelpers['getServerTransport'];
+    const { addPostTestCleanup, postTestCleanup } = createPostTestCleanups();
+    beforeEach(async () => {
+      testHelpers = await transport.setup({ client: opts, server: opts });
+      getClientTransport = testHelpers.getClientTransport;
+      getServerTransport = testHelpers.getServerTransport;
+      return async () => {
+        await postTestCleanup();
+        await testHelpers.cleanup();
+      };
+    });
+
+    test('handshake grace period of 0 should lead to closed connections', async () => {
+      const schema = Type.Object({
+        kept: Type.String(),
+        discarded: Type.String(),
+      });
+      const get = vi.fn(async () => ({ kept: 'kept', discarded: 'discarded' }));
```

Comment:
I understand you copied these from another test, but that other test was actually testing metadata stuff. I'd just remove the data here as it was confusing to me and I thought it was relevant. i.e. 
```suggestion
      const schema = Type.Object({
        kept: Type.String(),
        discarded: Type.String(),
      });
      const get = vi.fn(async () => ({ kept: 'kept', discarded: 'discarded' }));
```
```suggestion
      const schema = Type.Unknown();
      const get = vi.fn(async () => ({}));
```